ingeniamotion/configuration.py

```diff
@@ -930,11 +930,15 @@ def restore_tcp_ip_parameters(self, servo: str = DEFAULT_SERVO) -> None:
         drive.restore_tcp_ip_parameters()
 
     @overload
-    def get_mac_address(self, *, string_format: Literal[True]) -> str:
+    def get_mac_address(self, servo: str = ..., string_format: Literal[False] = False) -> int:
         ...
 
     @overload
-    def get_mac_address(self, *, string_format: Literal[False]) -> int:
+    def get_mac_address(self, servo: str, string_format: Literal[True]) -> str:
+        ...
+
+    @overload
+    def get_mac_address(self, *, string_format: Literal[True]) -> str:
```

Comment:
Why servo param is different in the three cases? Servo param is always the same type

Reply:
https://mypy.readthedocs.io/en/stable/more_types.html
"The default values of a function’s arguments don’t affect its signature – only the absence or presence of a default value does. So in order to reduce redundancy, it’s possible to replace default values in overload definitions with ... as a placeholder"

I think it's not necessary to write DEFAULT_SERVO
But at least it should contain = ...
The rest of arguments should be the same signature or more concrete (literal)

I'm updating the PR so it's not stuck forever

Reply:
Like it's now it's working with this simple typing test:
![image](https://github.com/user-attachments/assets/0a0ff9b8-f9cd-4146-93e8-04efc66f4eef)

Reply:
I thought this would work:
````
   @overload
    def get_mac_address(self, servo: str = ..., string_format: Literal[False] = ...) -> int:
        ...

    @overload
    def get_mac_address(self, servo: str = ..., string_format: Literal[True] = ...) -> str:
        ...
```

But: 
ingeniamotion\configuration.py:935: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]